### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by benmccann**
This keeps the project current with the latest version of dependencies. This has some small benefits. E.g. it is better to upgrade sooner than later because it means that no one will add code that is compatible only with old versions of dependencies making later upgrades more difficult

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,17 +40,17 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "eslint": "^4.0.0",
+    "eslint": "^4.7.0",
```
Comment (shinnn) [inline package.json]: `^4.0.0` covers `4.7.0`.

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ git:
 language: node_js
 node_js:
   - node
-  - '4'
+  - 'lts/*'
```
Comment (shinnn) [inline .travis.yml]: Don't include unrelated changes.

Comment (benmccann) [reply .travis.yml]: It's not unrelated. node.js 4 is really old and doesn't work with current versions of dependencies

File: package.json (hunk 1 above)
Comment (benmccann) [reply package.json]: Sure. But it doesn't hurt to specify `4.7.0` here does it? `ncu` automatically updates all dependencies. It's just a bit easier to update everything

Comment (shinnn) [reply package.json]: "doesn't hurt" is not enough convincing reason to accept the change.

**Comment by shinnn** [comment]
I repeat https://github.com/adametry/gulp-eslint/pull/207#pullrequestreview-63235919,

> Can you write a detailed description of this PR to tell us the purpose of these changes?

**Comment by benmccann** [comment]
I've reverted the non-major upgrades

I think it is helpful to stay on recent versions of dependencies. If you don't find value in this PR then feel free to close it
